config_cloud.go

```diff
@@ -213,24 +258,16 @@ func (p *ECSCapacityProvider) Validate() error {
 	catcher.NewWhen(p.Name == "", "must provide a capacity provider name")
 	catcher.Add(p.OS.Validate())
 	catcher.Add(p.Arch.Validate())
-	return catcher.Resolve()
-}
-
-// ECSArch represents a CPU architecture that can run containers in ECS.
-type ECSArch string
-
-const (
-	ECSArchAMD64 = "amd64"
-	ECSArchARM64 = "arm64"
-)
-
-func (a ECSArch) Validate() error {
-	switch a {
-	case ECSArchAMD64, ECSArchARM64:
-		return nil
-	default:
-		return errors.Errorf("unrecognized ECS capacity provider arch '%s'", a)
+	if p.OS == ECSOSWindows {
+		if p.WindowsVersion == "" {
+			catcher.New("must specify a particular Windows version when using Windows OS")
```

Comment:
Can we add this case for the empty string to `Validate` and remove the if/else structure here?

Reply:
We don't want empty string validation to pass if the OS is Windows, so I'm inclined to not allow the empty string unconditionally. This is not particularly complex logic anyways.